diff --git a/node_modules/react-native-date-picker/android/src/main/java/com/henninghall/date_picker/Utils.java b/node_modules/react-native-date-picker/android/src/main/java/com/henninghall/date_picker/Utils.java
index 5cf06e3..f28976c 100644
--- a/node_modules/react-native-date-picker/android/src/main/java/com/henninghall/date_picker/Utils.java
+++ b/node_modules/react-native-date-picker/android/src/main/java/com/henninghall/date_picker/Utils.java
@@ -32,7 +32,13 @@ public class Utils {
         if(dateString == null) return null;
         try {
             Calendar calendar = Calendar.getInstance(timeZone);
-            calendar.setTime(getIsoUTCFormat().parse(dateString));
+            // Check if the string is a numeric timestamp (positive or negative)
+            if (dateString.matches("-?\\d+")) {  // Matches both positive and negative numbers
+                long timestamp = Long.parseLong(dateString);
+                calendar.setTimeInMillis(timestamp);
+            } else {
+                calendar.setTime(getIsoUTCFormat().parse(dateString));
+            }
             return calendar;
         } catch (ParseException e) {
             e.printStackTrace();
diff --git a/node_modules/react-native-date-picker/src/DatePickerAndroid.js b/node_modules/react-native-date-picker/src/DatePickerAndroid.js
index a8a1e71..a038bef 100644
--- a/node_modules/react-native-date-picker/src/DatePickerAndroid.js
+++ b/node_modules/react-native-date-picker/src/DatePickerAndroid.js
@@ -96,7 +96,7 @@ const toIsoWithTimeZoneOffset = (date) => {
   /** @ts-ignore */
   if (!date) return undefined
   /** @ts-ignore */
-  return date.toISOString()
+  return date.getTime().toString()
 }
 
 /** @param {string} timestamp */
